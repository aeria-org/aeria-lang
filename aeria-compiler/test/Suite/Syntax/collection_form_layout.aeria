collection CollectionFormLayoutTest {
  formLayout {
    fields {
      prop1 {
        if @cond(truthy _id)
        verticalSpacing 3.0
        separator "top"
        span 3.0
        component {
          name "componentTest"
          props @js () => {
            return {
              prop1: "value1",
              prop2: "value2",
            }
          }
          @end
        }
      }
    }
  }
}
