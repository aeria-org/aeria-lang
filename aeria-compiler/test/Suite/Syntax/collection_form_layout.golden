{
  "kind": "Program",
  "collections": [
    {
      "name": {
        "value": "CollectionFormLayoutTest",
        "kind": "CollectionName"
      },
      "kind": "Collection",
      "formLayout": [
        {
          "verticalSpacing": 3,
          "span_": 3,
          "separator": "top",
          "name": {
            "value": "prop1",
            "kind": "PropertyName"
          },
          "kind": "LayoutItem",
          "if_": {
            "kind": "Cond",
            "expr": {
              "kind": "ETruthy",
              "expr": {
                "kind": "ELiteral",
                "expr": {
                  "value": {
                    "value": "_id",
                    "kind": "PropertyName"
                  },
                  "kind": "LProperty"
                }
              }
            }
          },
          "component": {
            "props": {
              "macro": "{\n            return {\n              prop1: \"value1\",\n              prop2: \"value2\",\n            }\n          }\n          ",
              "kind": "Macro"
            },
            "name": "componentTest",
            "kind": "LayoutItemComponent"
          }
        }
      ]
    }
  ]
}
